src/main/pages/che-7/extensions/assembly_telemetry.adoc

```diff
@@ -0,0 +1,260 @@
+
+We will then create a `settings.xml` in the repository root and add the coordinates and token to the `che-incubator` packages:
+
+.settings.xml
+[source,xml]
+----
+include::examples/telemetry/settings.xml[]
+----
+
+This file will be used when packaging our application in a container.  If you are running this locally, you can add the information to your personal `settings.xml` file.
+
+== Running the application
+
+The easiest way to run and test the application is in a {prod-short} workspace.  Run the application with
+
+```
+mvn quarkus:dev -Dquarkus.http.port=${CHE_WORKSPACE_TELEMETRY_BACKEND_PORT}
+```
+
+Note that if {prod-short} is secured using a self-signed certificate, you will need to add the certificate to a trust store and mount it into the workspace.  You will also need to add the java system property `-Djavax.net.ssl.trustStore=/path/to/trustStore` to the `mvn` command.  
+For example, assuming your trust store is located in `$JAVA_HOME/jre/lib/security/cacerts`:
+
+`keytool -import -alias self-signed-certificate -file path/to/self-signed-certificate -keystore $JAVA_HOME/jre/lib/security/cacerts`
+
+Followed by
+
+`mvn quarkus:dev -Dquarkus.http.port=${CHE_WORKSPACE_TELEMETRY_BACKEND_PORT} -Djavax.net.ssl.trustStore=$JAVA_HOME/jre/lib/security/cacerts`
+
+== Creating a concrete implementation of AnalyticsManager and adding specialized logic
+
+Create two new files in your project, `AnalyticsManager.java`, and `MainConfiguration.java`:
+
+.AnalyticsManager.java
+[source,java]
+----
+include::examples/telemetry/AnalyticsManagerSkeleton.java[]
+----
+
+`AnalyticsManager.java` is where we will put our logic specific to our telemetry system.
+
+.MainConfiguration.java
+[source,java]
+----
+include::examples/telemetry/MainConfiguration.java[]
+----
+
+`MainConfiguration.java` is the main entrypoint that creates an instance of `AnalyticsManager` and starts listening for events.
+
+== Implementing `isEnabled()`
+
+For our demo purpose, this method will just return true whenever it is called.  Whenever the server is running, we can safely assume it is enabled and operational.  
+
+.AnalyticsManager.java
+[source,java]
+----
+include::examples/telemetry/isEnabled.java[]
+----
+
+You can put more complex login in `isEnabled()`, however.  Perhaps your service should not be considered operational in certain cases.  For example, the link:https://github.com/redhat-developer/che-workspace-telemetry-woopra-backend/blob/465286d229fb43366c335d0c3379a803cd3cc6ad/src/main/java/com/redhat/che/workspace/services/telemetry/woopra/AnalyticsManager.java#L174[hosted {prod-short} woopra backend] checks that a configuration property exists before determining if the backend is enabled.
```

Comment:
looks like the link point to the precise commit, not sure if this is desired. Maybe it would be better to point to a branch?

Reply:
Fixed